core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -261,16 +259,25 @@ private void loginAndTakeBack(StaplerRequest req, StaplerResponse rsp, User u) t
     }
 
     /**
-     * Creates an user account. Used by admins.
+     * Creates a user account. Used by admins.
      *
      * This version behaves differently from {@link #doCreateAccount(StaplerRequest, StaplerResponse)} in that
      * this is someone creating another user.
      */
     public void doCreateAccountByAdmin(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        createAccountByAdmin(req, rsp, "addUser.jelly", "."); // send the user back to the listing page on success
+    }
+
+    /**
+     * Creates a user account. Requires {@link Jenkins#ADMINISTER}
+     */
+    public User createAccountByAdmin(StaplerRequest req, StaplerResponse rsp, String addUserView, String successView) throws IOException, ServletException {
         checkPermission(Jenkins.ADMINISTER);
-        if(createAccount(req, rsp, false, "addUser.jelly")!=null) {
-            rsp.sendRedirect(".");  // send the user back to the listing page
+        User u = createAccount(req, rsp, false, addUserView);
```

Comment:
... this is really not good, but given that the createAccount method handles changing views, I'm not sure if this could be made a lot better without more significant refactorings